newIDE/app/src/MainFrame/Editors/StartPage/index.js

```diff
@@ -160,7 +238,8 @@ class StartPage extends BaseEditor {
                     }
                   />
                 </Line>
-                <Line alignItems="center">
+                {/*TODO Le className disparait au rendu*/}
+                <Line alignItems="center" className="socialNetwork">
```

Comment:
Line is a component, not a DOM element, so className is passed as a prop to "Line", then unused in the render method of Line. So it's not appearing anywhere. Flow will yell at you for passing a prop that is not recognized (`npm run flow`) (much like what would happen in TypeScript).

I don't think it's a good idea to add identifiers on "Lines" anyway. They are component that are used to describe the layout/structure of the UI. Instead, we can do two things:
1. Add a "identifier" prop on some common UI components (mainly the buttons and icons). Then, in their `render` method, pass the prop "identifier" to the material-ui component (and material-ui will hopefully pass it to "divs" and other DOM elements). Or if the `render` of the component is showing a div around some material-ui components, you can also put the identifier here.
2. We can also make a `GuidelineMarker` component that is just a div that render the children, with the identifier. Something like:

```
const GuidelineMarker = ({ identifier, children }) => <div className={"guideline-" + identifier}>{children}</div>
```

You'll see that I've prefixed the className with "guideline-", to avoid risking an already existing CSS class to be added to this div. Note that I'm not a super big fan of this "extra div", as it will probably break the layout of components... so we should aim at adding such "identifier" props to a few selected UI components I guess, like RaisedButton, FlatButton, icons (see most of them in the `newIDE/app/src/UI` folder)

Reply:
Additional note again, I've used "className" everywhere as an example, but I would prefer a "data-guideline-identifier" in the final solution :) className should only be for the styling.